src/express/HttpServer.ts

```diff
@@ -0,0 +1,137 @@
+          const errorMessage = error instanceof Error ? error.message : 'An unknown error occurred';
+          return res.status(500).json({ error: errorMessage });
+        }
+      });
+
+      this.app.get('/chain/:chain/metadata', async (req, res) => {
+        try {
+          const chainName = req.params.chain as ChainName;
+          const metadata = await registry.getChainMetadata(chainName);
+          if (!metadata) {
+            return res.status(404).json({ error: 'Chain metadata not found' });
+          }
+          return res.json(metadata);
+        } catch (error) {
+          const errorMessage = error instanceof Error ? error.message : 'An unknown error occurred';
+          return res.status(500).json({ error: errorMessage });
+        }
+      });
+
+      this.app.post('/chain/:chain/metadata', async (req, res) => {
+        try {
+          const chainName = ZChainName.parse(req.params.chain);
+          const metadata = ChainMetadataSchema.parse(req.body);
+          await registry.updateChain({
+            chainName,
+            metadata,
+          });
+          return res.status(200).json({ message: 'Chain metadata updated successfully' });
+        } catch (error) {
+          const errorMessage = error instanceof Error ? error.message : 'An unknown error occurred';
+          return res.status(500).json({ error: errorMessage });
+        }
+      });
+
+      this.app.get('/chain/:chain/addresses', async (req, res) => {
+        try {
+          const chainName = req.params.chain as ChainName;
+          const addresses = await registry.getChainAddresses(chainName);
+          if (!addresses) {
+            return res.status(404).json({ error: 'Chain addresses not found' });
+          }
+          return res.json(addresses);
+        } catch (error) {
+          const errorMessage = error instanceof Error ? error.message : 'An unknown error occurred';
+          return res.status(500).json({ error: errorMessage });
+        }
+      });
+
+      this.app.get('/warp-route/:id', async (req, res) => {
+        try {
+          const id = req.params.id as WarpRouteId;
+          const warpRoute = await registry.getWarpRoute(id);
+          if (!warpRoute) {
+            return res.status(404).json({ error: 'Warp route not found' });
+          }
+          return res.json(warpRoute);
+        } catch (error) {
+          const errorMessage = error instanceof Error ? error.message : 'An unknown error occurred';
+          return res.status(500).json({ error: errorMessage });
+        }
+      });
+
+      const server = this.app.listen(port, () =>
+        this.logger.info(`Server running on port ${port}`),
+      );
+
+      server.on('request', (req, _res) => this.logger.info('Request:', req.url));
+
+      server.on('error', (error) => this.logger.error('Server error:', error));
+    } catch (error) {
+      this.logger.error('Error starting server:', error);
+      process.exit(1);
+    }
```

Comment:
Can we have the endpoint methods implemented somewhere else as a function and register the endpoints and handlers using a method called `registerEndpoint` or something similar? It would make checking the endpoint implementations a bit easier and will make the `start` method smaller